src/auto_scheduler/search_policy/sketch_policy_rules.h

```diff
@@ -201,6 +201,26 @@ class InitVectorization : public InitPopulationRule {
   ResultKind Apply(SketchPolicyNode* policy, State* state) const final;
 };
 
+/********** Mutation **********/
+
+/*! \brief The base class for mutation rules used in the evolutionary search. */
+class MutationRule : public InitPopulationRule {
```

Comment:
Inherit from this class seems strange, we can just have an independent class.

The updated implementation in the last PR is trying to use a macro definition format:
https://github.com/apache/incubator-tvm/pull/6269/files#diff-27486a3e525b590252f09c3fca9d2718

Reply:
Please also take a look at `MutateComputeLocation` added by the latest commit. The required functions is exactly the same as the one used in initial population, so we should be able to make up a general rule that includes both initial population and mutation.